src/core/unit.c

```diff
@@ -4279,7 +4279,6 @@ int unit_patch_contexts(Unit *u) {
                          * UID/GID around in the file system or on IPC objects. Hence enforce a strict
                          * sandbox. */
 
-                        ec->private_tmp = true;
```

Comment:
i'd still imply this, it's prettier that way, because it tells external tools that we do not share the regular /tmp/ on these services

Reply:
I removed it to avoid triggering all the machinery that sets up the directories in the host filesystem, passes the pipes, adds the tracking, etc since it's not needed

Reply:
And actually given the exact semantics on PrivateTmp, where the storage is actually on the host and it is visible to root there, and it can be shared, I think we should not toggle this, because it would not correspond to what actually happens, and suddenly it could mean either the new thing or the old thing.

Reply:
sorry, i am still of the opinion that the knob should abstract things, i.e. expose this as true if /tmp is not shared with the host.

It doesn't matter too much how we implement this. Key really is that the knob should be helpful if people try to figure out if stuff they put in /tmp on the host is going to be visible in the service or not. I.e. focus on the stuff that matters to users, not on the implementation details.